Code/Tools/CrashHandler/Uploader/src/CrashUploader.cpp

```diff
@@ -199,6 +199,29 @@ namespace O3de
             {nullptr, no_argument, nullptr, EndFlags}
         };
 
+        for (int i = 0; i < argc; ++i)
+        {
+            std::string inputString = argv[i];
+            size_t equalPos = inputString.find_first_of('=');
+            if (equalPos == std::string::npos || equalPos == inputString.length() - 1)
+            {
+                continue;
+            }
+
+            std::string tokenStr = inputString.substr(0, equalPos);
+            std::string valueStr = inputString.substr(equalPos + 1);
+
+            if (tokenStr == "--submission-token")
+            {
+                m_submissionToken = valueStr;
+            }
+            else if (tokenStr == "--executable-name")
+            {
+                m_executableName = valueStr;
+            }
+        }
+
+        // FIXME: argument parsing is broken, fallback above
```

Comment:
What is broken here, exactly? AZCore does have a CommandLine class that can parse command lines, and has been pretty well tested.  It can only be used if the AzCore memory system is active though.  At the least, indicate why its broken for future maintainer?

Is it maybe trying to avoid any AzCore memory stuff in case its corrupt?  But most crash handlers run as out-of-process satellite executables for that same reason, the crashing app can't be trusted.

Reply:
Hey ! getopt_long is a CrashPad library call, I haven't looked further yet as I want to update our version in order to support linux and android.
I believe it is needed to keep crashpad calls as it might be initializing things in the library